### PR TITLE
Feature: filter posting journal by amount

### LINKED ISSUE #1354: (Feature Request) Add a search field for amount to the Posting Journal
We've received feedback asking for us to put a search by amount field in the Posting Journal search modal.  This should be easily accomplished used FilterParser

### PR DESCRIPTION
**Pull request by jniles**
This PR implements filtering by amount in the Posting Journal.  It also makes sure the journal is using `FilterParser` on the server and refactors the journal search modal to use `bhAccountSelect`.

In order to accomplish this, I made a new `FilterParser` method to take in multiple query parameters.  It's useful in the following case:
```js
filters.customMultiParameters('amount', '(journal.credit_equiv = ? OR journal.debit_equiv = ?)', [opts.amount, opts.amount]);
```

Closes #1354.

---
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
Closing this while #1359 is developed and applied here.

**Comment by sfount** [comment]
@jniles could we merge this first and then this design/ development can take place after? This PR looks good to me and contains some other good refactors.

**Comment by jniles** [comment]
Cheers!

File: server/controllers/finance/journal/index.js [hunk 1]
```diff
@@ -129,88 +110,31 @@ function find(options) {
...
-  }
-
-  // filter on reference uuid
-  if (options.reference_uuid) {
-    const referenceUuid = db.bid(options.reference_uuid);
-    conditions.statements.push('p.reference_uuid = ?');
-    conditions.parameters.push(referenceUuid);
-    delete options.reference_uuid;
-  }
+  filters.dateFrom('dateFrom', 'trans_date');
```
Comment (sfount) [inline server/controllers/finance/journal/index.js:115]: 👍

File: server/lib/filter.js [hunk 2]
```diff
@@ -176,13 +176,24 @@ class FilterParser {
 
   custom(filterKey, preparedStatement, preparedValue) {
     if (this._filters[filterKey]) {
-      let searchValue = preparedValue || this._filters[filterKey];
-
+      const searchValue = preparedValue || this._filters[filterKey];
```
Comment (sfount) [inline server/lib/filter.js:179]: Good catch!

**Review by sfount — APPROVED** [review]

**Pull request merged**